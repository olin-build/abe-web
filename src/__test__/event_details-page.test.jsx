import moment from 'moment';
import React from 'react';
import renderer from 'react-test-renderer';
import EventDetailsPage from '../pages/details/event-details-page';

describe('EventDetailsPage', () => {
  test('loading', () => {
    const component = renderer.create(<EventDetailsPage eventData={null} setSidebarMode={() => undefined} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('regular event', () => {
    const event = {
      title: 'Title',
      location: 'Location',
      description: 'Description',
      start: moment('2018-05-07T09:00:00-04:00'),
      end: moment('2018-05-07T10:00:00-04:00'),
      allDay: false,
    };
    const component = renderer.create(<EventDetailsPage eventData={event} setSidebarMode={() => undefined} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('single-day all-day event', () => {
    const event = {
      title: 'Title',
      location: 'Location',
      description: 'Description',
      start: moment('2018-05-07T00:00:00-04:00'),
      end: moment('2018-05-07T23:59:59-04:00'),
      allDay: true,
    };
    const component = renderer.create(<EventDetailsPage eventData={event} setSidebarMode={() => undefined} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('multi-day all-day event', () => {
    const event = {
      title: 'Title',
      location: 'Location',
      description: 'Description',
      start: moment('2018-05-07T00:00:00-04:00'),
      end: moment('2018-05-08T23:59:59-04:00'),
      allDay: true,
    };
    const component = renderer.create(<EventDetailsPage eventData={event} setSidebarMode={() => undefined} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
