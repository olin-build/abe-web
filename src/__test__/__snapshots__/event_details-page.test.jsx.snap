// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetailsPage loading 1`] = `
<div
  className="row expanded page-container"
>
  <div
    className="row content-container"
  >
    <h1
      className="page-title"
    >
      Loading...
    </h1>
  </div>
</div>
`;

exports[`EventDetailsPage multi-day all-day event 1`] = `
<div
  className="row expanded page-container"
>
  <div
    className="row content-container"
  >
    <h1
      className="page-title"
    >
      Title
    </h1>
    <div
      className="event-info-container"
    >
      <div
        className="event-date-location-container"
      >
        <span
          className="event-start"
        >
          Mon, May 7, 2018
        </span>
        <span>
           to
          <span
            className="event-start"
          >
             Tue, May 8, 2018
          </span>
        </span>
        <p
          className="event-location"
        >
          Location
        </p>
      </div>
      <div
        className="description-container"
      >
        <p>
          Description
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`EventDetailsPage regular event 1`] = `
<div
  className="row expanded page-container"
>
  <div
    className="row content-container"
  >
    <h1
      className="page-title"
    >
      Title
    </h1>
    <div
      className="event-info-container"
    >
      <div
        className="event-date-location-container"
      >
        <span
          className="event-start"
        >
          Mon, May 7, 2018 9:00 AM
        </span>
        <span>
           to
          <span
            className="event-start"
          >
             10:00 AM
          </span>
        </span>
        <p
          className="event-location"
        >
          Location
        </p>
      </div>
      <div
        className="description-container"
      >
        <p>
          Description
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`EventDetailsPage single-day all-day event 1`] = `
<div
  className="row expanded page-container"
>
  <div
    className="row content-container"
  >
    <h1
      className="page-title"
    >
      Title
    </h1>
    <div
      className="event-info-container"
    >
      <div
        className="event-date-location-container"
      >
        <span
          className="event-start"
        >
          Mon, May 7, 2018
        </span>
        <p
          className="event-location"
        >
          Location
        </p>
      </div>
      <div
        className="description-container"
      >
        <p>
          Description
        </p>
      </div>
    </div>
  </div>
</div>
`;
